subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/DefaultJavaPluginExtension.java

```diff
@@ -122,6 +127,17 @@ public ModularitySpec getModularity() {
         return modularity;
     }
 
+    @Override
+    public JavaToolchainSpec getToolchain() {
+        return toolchain(Actions.doNothing());
```

Comment:
Why not return `toolchain` directly?

Reply:
Mostly consistency if toolchain() evolves (it had more logic in it before). Simplified.